Finalizado poder de 'Saboteur'

Trosdan opened the pull request:
fix #1 

- Saboteur

Trosdan committed:
feat #1 - Finalizado poder de 'Saboteur'

Trosdan committed:
Merge branch 'develop' into feature/powersSaboteur

pinechain commented on Assets/Scripts/Board/BoardController.cs:
```diff
@@ -17,7 +17,8 @@ public int CreatedLines
 
         // Scripts References
-        BoardGenerate boardGenerate;
+        [HideInInspector]
+        public BoardGenerate boardGenerate;
 
         [Header("Board Settings")]
         public BoardStatusTypes boardStatusTypes;
```
Devemos evitar o uso de atributos públicos em classes, para aumentar a segurança e a confiabilidade das classes. Isso também contribui para restringir quem acessa ou modifica cada atributo, facilitando o debug. Existem alguns motivos para usar atributos públicos, que podem ser cobertos por alternativas:

- Mostrar e modificar o atributo no Inspector;
  - O atributo pode ser privado e marcado com [[SerializeField]](https://docs.unity3d.com/ScriptReference/SerializeField.html), presente no UnityEngine.
- Ser acessado e alterado por classes de fora;
  - O atributo pode ser privado e [usado como _backing field_ em uma propriedade](https://docs.microsoft.com/pt-br/dotnet/csharp/programming-guide/classes-and-structs/properties), ou um método, permitindo a implementação de regras especiais de acesso e/ou validações de valores.

pinechain commented on Assets/Scripts/Player/Power/PowerExecutor.cs:
```diff
@@ -108,11 +108,19 @@ public int getTopmostLineIndex()
             boardController.destroyLine(breakableLineIndex);
         }
 
+        public void createLine(int createLineIndex) {
+
+        }
+        
```
Código vestigial?

Trosdan replied:
Sera removido.

pinechain commented on Assets/Scripts/Power/Domain/Saboteur.cs:
```diff
@@ -0,0 +1,35 @@
+        override public IEnumerator execute(PowerExecutor executor)
+        {
+            executor.createObstacle();
+            yield return new WaitForSeconds(0);
```
Esse valor do WaitForSeconds pode ser parametrizado para permitir que esse poder tenha uma certa "animação" na hora de criar mais de um obstáculo, similar ao que acontece com o poder LineBreaker atualmente. Entretanto isso depende do que for especificado para o poder.

Invoco @nuneeez para pesar essa decisão.

Trosdan replied:
Irei implementar a criação de mais obstaculo de acordo como nivel e o WaitForSeconds vai ser usado no tempo de criação dos obstaculos

pinechain commented on Assets/Scripts/Power/Domain/Saboteur.cs:
```diff
@@ -0,0 +1,35 @@
+
+        override public IEnumerator execute(PowerExecutor executor)
+        {
+            executor.createObstacle();
```
O poder cria mais de um obstáculo quando criado no nível 2 ou 3?

Trosdan replied:
Sim, sera implementado.

Trosdan committed:
Alterações referente ao MergeRequest #15

 - Privacidade da variavel boardGenerate\n - Removido função sem uso\n - Poder sendo ativado de acordo com o nivel

Trosdan committed:
Merge branch 'feature/powersSaboteur' of https://github.com/Design-Under-Konstruction-Studio/Space-Parrrty into feature/powersSaboteur

pinechain commented on Assets/Scripts/Power/Domain/Saboteur.cs:
```diff
@@ -0,0 +1,37 @@
+    {
+        [Header("Internal state - do not assign")]
+        [SerializeField]
+        private int quantityObstacleLimit = 2;
```
Atributo não está sendo usado, favor remover.

Trosdan committed:
Removido variavel inutilizável (quantityObstacleLimit)
